gobblin-compaction/src/main/java/gobblin/compaction/dataset/Dataset.java

```diff
@@ -231,10 +230,8 @@ public double reducePriority(double reductionFactor) {
     return this.priority;
   }
 
-  public void checkIfNeedToRecompact(long lateDataCount, long nonLateDataCount) {
```

Comment:
Since this is a public method we still need to maintain backward compatibility. You can mark it as deprecated.